### PR TITLE
adding -e --entities flag

### PR DESCRIPTION
**Pull request by Public-Health-Bioinformatics**
Enables saving of XML ontology formats to encode the ontology’s
entities in the output.

### CODE REVIEW

**Comment by jamesaoverton** [comment]
Much appreciated! (Actually, it's hard to express just how much we appreciate PRs.)

I have two requests:

1. Can you please add a test, to help us avoid accidentally breaking this code. My suggestion would be a round-trip test in `IOHelperTest.java`: load one of the fancier test files (maybe `mireot.owl`), save as XML with entities, load that XML with entities, then diff against the original to make sure that they are the same.

2. I'd like to change some of the names. I'm worried about confusion between XML entity, `OWLEntity`, and BFO `entity`, and I'd prefer to leave the short `-e` option available for use by other commands. I'll make some line-comments about this.

(See #114)

File: robot-command/src/main/java/org/obolibrary/robot/CommandLineHelper.java [hunk 1]
```diff
@@ -538,6 +540,7 @@ public static Options getCommonOptions() {
         o.addOption("p", "prefix",  true,  "add a prefix 'foo: http://bar'");
         o.addOption("P", "prefixes", true, "use prefixes from JSON-LD file");
         o.addOption("noprefixes", false, "do not use default prefixes");
+        o.addOption("e", "entities", false, "use entity substitution when saving XML ontology");
```
Comment (jamesaoverton) [inline robot-command/src/main/java/org/obolibrary/robot/CommandLineHelper.java]: I'd prefer to use a single long option for this: `--xml-entities`

File: robot-core/src/main/java/org/obolibrary/robot/IOHelper.java [hunk 2]
```diff
@@ -683,6 +689,29 @@ public void setContext(Map<String, Object> map) {
     }
 
     /**
+     * Set whether or not XML entities will be swapped into URI's in saveOntology XML output formats.
+     *
+     * @param map a map of strings for the new JSON-LD context
+     */
+    public void setEntityFlag(Boolean entityFlag) {
```
Comment (jamesaoverton) [inline robot-core/src/main/java/org/obolibrary/robot/IOHelper.java]: Please rename to `setXMLEntityFlag`

File: robot-core/src/main/java/org/obolibrary/robot/IOHelper.java [hunk 3]
```diff
@@ -683,6 +689,29 @@ public void setContext(Map<String, Object> map) {
...
+            logger.warn(e.getMessage());
+        }
+    }
+
+    /**
+     * Get the useXMLEntities flag.
+     *
+     * @return boolean useXMLEntities flag
+     */
+    public Boolean getEntityFlag() {
```
Comment (jamesaoverton) [inline robot-core/src/main/java/org/obolibrary/robot/IOHelper.java]: Please rename to `getXMLEntityFlag`

File: robot-command/src/main/java/org/obolibrary/robot/CommandLineHelper.java (hunk 1 above)
Comment (Public-Health-Bioinformatics) [inline robot-command/src/main/java/org/obolibrary/robot/CommandLineHelper.java]: Good point.  Will implement all this shortly.

**Comment by Public-Health-Bioinformatics** [comment]
Inside pull request I accidentally deleted an examples/uberon_mireot.owl file.  Its restored, and all should be working and tested.

**Comment by jamesaoverton** [comment]
Looks good. Your tempFile solution is a good idea.

The only problem is that we should not include the binary files in our source tree. I'm not sure how to remove those from the pull request's history. So I'm going to start a new branch with your source changes but without the binary files, and merge that.

**Comment by jamesaoverton** [comment]
Thanks again! This is a great option to have, and I'm really glad that you implemented it.

I merged these changes in 8983128670790a61d734eb32f289ab178d9ebc24, and in f54dc017c790a6de41ecaf91662135d848cb183b I updated the `.gitignore` to stop the binary files from being accidentally committed in the future.

**Comment by Public-Health-Bioinformatics** [comment]
Great.  It was a little bit of a java / owllib learning experience, glad it proved fruitful in the end.  I much prefer reading .owl with entities in place.

**Pull request closed without merging**